### PR TITLE
eliminate use of netty bytebuf in logdata

### PR DESCRIPTION
**Pull request by no2chem**
This patch replaces the Netty bytebuf in LogData with a standard byte array, reducing the dependency on 
Netty bytebufs throughout the code

### CODE REVIEW

File: runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java [hunk 1]
```diff
@@ -74,18 +79,26 @@ public LogData(DataType type) {
         this.metadataMap = new EnumMap<>(IMetadata.LogUnitMetadataType.class);
     }
 
-    public LogData(DataType type, ByteBuf buf) {
+    public LogData(final DataType type, final ByteBuf buf) {
```
Comment (Maithem) [inline runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java]: Why not change the second parameter to byte[]?

Comment (no2chem) [inline runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java]: ah, I see. I wanted this change to be small, changing the method signature would have changed everything else that created a LogData. I can take care of that too.

**Comment by no2chem** [comment]
@Maithem - I cleaned up the addressSpace read cache (by creating InMemoryLogData), but I've left LogUnitClient etc. alone - I think we should save that for another PR, which will involve fixing the serialization scheme. I'm trying to keep PRs manageable - cleaning up serialization will definitely touch large parts of the code base, unfortunately. 

If you're okay with this part of the PR, please approve.

**Comment by Maithem** [comment]
@no2chem looks fine, but I think you have compilation errors.

**Comment by no2chem** [comment]
@Maithem Hm? The travis build passed though?

**Comment by Maithem** [comment]
@no2chem I see this : https://travis-ci.org/CorfuDB/CorfuDB/builds/189133029

**Pull request merged**